MAVProxy/mavproxy.py

```diff
@@ -861,6 +861,34 @@ def run_script(scriptfile):
             mpstate.console.writeln("-> %s" % line)
         process_stdin(line)
     f.close()
+    
+def set_mav_version(mav10, mav20, autoProtocol, mavversion):
+    '''Set the Mavlink version based on commandline options'''
+    if(mav10 == True or mav20 == True or autoProtocol == True):
+        print("Warning: Using depreciated --mav10, --mav20 or --auto-protocol options. Use --mavversion instead")
+    
+    #sanity check the options
```

Comment:
Unless you want different error messages here, this can be simplified.  Given only one of the options is set, you can increment a counter, and if that counter is >1 in the end you can throw the error.
e.g.
```
if maversion is not None:
  sources += 1
if mav10:
  sources += 1
.
.
```